### ISSUE TITLE
SugarRecord.delete search id column with Class.getDeclaredField

### ISSUE THREAD

**Issue by tlevavasseur** [satyan/sugar#655]
SugarRecord.delete(Object) searches the "id" column with Class.getDeclaredField then when the id field is in superclass you have this error message : "annotated object has no id"

**Comment by fxpage** [comment]
@tlevavasseur Have you fixed the issue?

**Comment by tlevavasseur** [comment]
Nope, just a workaround : abstract getter & setter in superclass and private attribute in each subclass.

**Comment by drake29a** [comment]
Hello, just one remark, when you save item, you iterate superclasses when looking for a field, so id field is found, and save can be done, problem occurs only when you try to delete saved record.
